### PR TITLE
UUID implementation for patient.

### PR DESCRIPTION
**Pull request by pithonsoft**
Signed-off-by: Pithonsoft <pithonsoft@gmail.com>

### CODE REVIEW

**Comment by pithonsoft** [comment]
Hi Brady,

Please review the code.

**Comment by bradymiller** [comment]
Just for reference purposes, this is an alternative implementation to the prior PR:
https://github.com/openemr/openemr/pull/2360

**Comment by bradymiller** [comment]
hi @pithonsoft , This is definitely a slimmed down approach to the prior patient uuid PR (https://github.com/openemr/openemr/pull/2360) .
@sjpadgett , @adunsulag , any thoughts on this versus the prior PR? Goal is to have a framework to basically convert all tables to using uuid in the future, which causes me to lean more towards the prior PR, but I honestly don't know the best route to take.

**Comment by sjpadgett** [comment]
I guess I wonder why the author @pithonsoft  wants to do this PR?

**Comment by pithonsoft** [comment]
Hi,

Per the discussion in the other thread what we understand that there is no need for maintaining centralized table for UUID and transfer the logic for ensuring the uniqueness to calling function.  On that basis we had implemented this.
Going through different articles we have a strong feeling that going for UUID as primary key for individual tables seriously affect speed especially UUID version 4.

File: sql/database.sql [hunk 1]
```diff
@@ -5376,6 +5376,7 @@ CREATE TABLE `patient_data` (
   `financial_review` datetime default NULL,
   `pubpid` varchar(255) NOT NULL default '',
   `pid` bigint(20) NOT NULL default '0',
+  `uuid` char(36) NOT NULL default '',
```
Comment (mdsupport) [inline sql/database.sql]: Per RFC 4122, uuid is 16 octets.  MySQL team has a good example in [this post](https://mysqlserverteam.com/mysql-8-0-uuid-support/).  8.0 mysql example is

> b BINARY(16)  DEFAULT (UUID_TO_BIN(UUID())),

**Comment by robertdown** [comment]
Why create a separate ID here? Is the thought that eventually UUID will replace PID?

Most EMRs have an MRN (Medical Record Number) and a number to identify an encounter (In Cerner it's called a FIN#). Thinking from a clinical perspective, I toss around FINs and MRNs throughout my shift a LOT. No way I could use a UUID for that. Now, there _very well may be_ a UUID behind the curtain in Cerner that I as an end user never see.

Maybe we do use a UUID (offline use is obviously number one reason) but that stays tucked behind and is only used in code and we continue to allow some kind of MRN (i.e. PID) and Encounter ID (EID??) for human consumption?

**Comment by mdsupport** [comment]
@robertdown, UUIDs like 'id'/auto-increment fields are not meant for user consumption. However unlike 'id', in case of this EMR, they could work great for offline or distributed use cases.  Converting system logic based on auto-increment PKs to UUID is not a trivial task .. specially for this application that has no central code implementing business logic.

@pithonsoft, not sure of your objectives for introducing this and if you plan to propagate change throughout the system.  In general it is not a good db practice to have multiple PKs for any table .. specially not a core table like patient_data.  But if there is a compelling case, least the project can do is get basic datatype correct.  Typical db engines do not like strings as indexes.

**Pull request closed without merging**